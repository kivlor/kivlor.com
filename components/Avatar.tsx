/** @jsxImportSource preact */

import { asset } from '$fresh/runtime.ts';

export default function Avatar() {
  return (
    <figure class={`m-auto w-32 md:w-40 h-32 md:h-40 bg-gray-200 dark:bg-gray-800 rounded-full`}>
      <img src={asset('/avatar.png')} alt="Kivlor" />
    </figure>
  );
}
