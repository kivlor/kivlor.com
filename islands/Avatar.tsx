/** @jsxImportSource preact */

import { asset } from "$fresh/runtime.ts";

import { hovering } from '../utils/state.ts';

export default function Avatar() {
  let background = 'bg-green-100';
  switch (hovering.value) {
    case 'github':
      background = 'bg-gray-200';
      break;
    case 'instagram':
      background = 'bg-pink-200';
      break;
    case 'linkedin':
      background = 'bg-blue-200';
      break;
  }

  return (
    <figure class={`m-auto rounded-full w-40 md:w-48 h-40 md:h-48 ${background} transition-colors`}>
      <img src={asset('/avatar.png')} alt="Kivlor" />
    </figure>
  );
}
