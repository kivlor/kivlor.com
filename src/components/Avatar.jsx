import * as React from 'react';

import avatarURL from '../images/avatar.gif';

export const Avatar = () => (
  <figure className="avatar">
    <img src={avatarURL} alt="Kivlor" />
  </figure>
);
