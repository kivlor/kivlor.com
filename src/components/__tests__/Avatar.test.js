import React from 'react';
import renderer from 'react-test-renderer';

import { Avatar } from '../Avatar';

describe('Avatar', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Avatar />);

    expect(component).toBeTruthy();
  });

  it('matches snapshot', () => {
    const component = renderer.create(<Avatar />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
