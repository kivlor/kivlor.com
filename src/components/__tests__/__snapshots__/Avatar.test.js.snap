// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches snapshot 1`] = `
<figure
  className="avatar"
>
  <img
    alt="Kivlor"
    src="test-file-stub"
  />
</figure>
`;
